lib/chef/resource/dsc_resource.rb

```diff
@@ -31,6 +31,7 @@ def initialize(name, run_context)
         super
         @properties = {}
         @resource = nil
+        @allow_reboot = :nothing

```

Comment:
yeah. I was initially confused by the booleanish name. `reboot_action` is more clear.